.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,28 @@
+---
+name: golangci-lint
+
+on:
+  push:
```

Comment:
I think this should run on all PRs, similar to the unit tests. Should we just have 1 workflow for both as in https://github.com/EvergenEnergy/sparkplughost/blob/main/.github/workflows/ci.yaml#L40?

Reply:
I like to have 2 steps between tests and linter. I like to have an early indication linting is broken